src/core/ext/filters/client_channel/resolver/google_c2p/google_c2p_resolver.cc

```diff
@@ -28,6 +28,9 @@ namespace grpc_core {
 
 namespace {
 
+std::string g_traffic_director_uri =
```

Comment:
The style guide prohibits global variables that are not trivially destructible.

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

In this case, I suggest making this a `char *` and having `GoogleCloud2ProdResolverInit()` set it to the result of `gpr_getenv()`.  Then the code in `StartXdsResolver()` can use `"directpath-trafficdirector.googleapis.com"` if `g_traffic_director_uri` is null.

Reply:
Oh, and `GoogleCloud2ProdResolverShutdown()` should free it.

Reply:
Whoops! Fixed.